heat/core/dndarray.py

```diff
@@ -248,6 +259,111 @@ def strides(self):
     def T(self):
         return linalg.transpose(self, axes=None)
 
+    @property
+    def array_with_halos(self):
+        return self.cathalo()
+
+    def __prephalo(self, start, end):
+        """
+        Extracts the halo indexed by start, end from self.array in the direction of self.split
+
+        Parameters
+        ----------
+        start : int
+            start index of the halo extracted from self.array
+        end : int
+            end index of the halo extracted from self.array
+        Returns
+        -------
+        halo : torch tensor
+            The halo extracted from self.array
+        """
+        ix = [slice(None, None, None)] * len(self.shape)
+        try:
+            ix[self.split] = slice(start, end)
+        except IndexError:
+            print("Indices out of bound")
+
+        return self.__array[ix].clone().contiguous()
+
+    def gethalo(self, halo_size):
```

Comment:
get_halo()?

Does this need to be public?

Reply:
why not, if someone needs to fetch a halo. Why should this not be public?

Reply:
okay, change to get_halo() would be good though. I am thinking whether a halo property would feel more Pythonic. What do you think?

Reply:
get_halo() fixed